tests/src/test/kotlin/se/ansman/kotshi/NestedSealedClassTest.kt

```diff
@@ -13,18 +13,18 @@ class NestedSealedClassTest {
     @Test
     fun nested1() {
         assertEquals(NestedSealedClass.Nested1.NestedChild1("nc1"), adapter.fromJson("""{"type":"nestedChild1","v":"nc1"}"""))
-        assertEquals("""{"v":"nc1"}""", adapter.toJson(NestedSealedClass.Nested1.NestedChild1("nc1")))
+        assertEquals("""{"type":"nestedChild1","v":"nc1"}""", adapter.toJson(NestedSealedClass.Nested1.NestedChild1("nc1")))
     }
 
     @Test
     fun nested2() {
         assertEquals(NestedSealedClass.Nested1.Nested2.NestedChild2("nc2"), adapter.fromJson("""{"type":"nestedChild2","v":"nc2"}"""))
-        assertEquals("""{"v":"nc2"}""", adapter.toJson(NestedSealedClass.Nested1.NestedChild1("nc2")))
+        assertEquals("""{"type":"nestedChild1","v":"nc2"}""", adapter.toJson(NestedSealedClass.Nested1.NestedChild1("nc2")))
     }
 
     @Test
     fun nested3() {
         assertEquals(NestedSealedClass.Nested3.NestedChild4("nc4"), adapter.fromJson("""{"type":"nested3","subtype":"nestedChild4","v":"nc4"}"""))
-        assertEquals("""{"v":"nc4"}""", adapter.toJson(NestedSealedClass.Nested3.NestedChild4("nc4")))
+        assertEquals("""{"subtype":"nestedChild4","v":"nc4"}""", adapter.toJson(NestedSealedClass.Nested3.NestedChild4("nc4")))
```

Comment:
hmm, should this have emitted `"type":"nested3"` as well?

Reply:
Yes, you'll need the traverse the whole hierarchy to achieve this.